crates/bevy_ecs/src/storage/table/column.rs

```diff
@@ -0,0 +1,705 @@
+    #[inline]
+    pub(crate) unsafe fn check_change_ticks(&mut self, len: usize, change_tick: Tick) {
+        for i in 0..len {
+            // SAFETY:
+            // - `i` < `len`
+            // we have a mutable reference to `self`
+            unsafe { self.added_ticks.get_unchecked_mut(i) }
+                .get_mut()
+                .check_tick(change_tick);
+            // SAFETY:
+            // - `i` < `len`
+            // we have a mutable reference to `self`
+            unsafe { self.changed_ticks.get_unchecked_mut(i) }
+                .get_mut()
+                .check_tick(change_tick);
+        }
+    }
+
+    /// Clear all the components from this column.
+    ///
+    /// # Safety
+    /// - `len` must match the actual length of the column
+    /// -   The caller must not use the elements this column's data until [`initializing`](Self::initialize) it again (set `len` to 0).
+    pub(crate) unsafe fn clear(&mut self, len: usize) {
+        self.added_ticks.clear_elements(len);
+        self.changed_ticks.clear_elements(len);
+        self.data.clear_elements(len);
+        #[cfg(feature = "track_change_detection")]
+        self.changed_by.clear_elements(len);
+    }
+
+    /// Because this method needs parameters, it can't be the implementation of the `Drop` trait.
+    /// The owner of this [`ThinColumn`] must call this method with the correct information.
+    ///
+    /// # Safety
+    /// - `len` is indeed the length of the column
+    /// - `cap` is indeed the capacity of the column
+    /// - the data stored in `self` will never be used again
+    pub(crate) unsafe fn drop(&mut self, cap: usize, len: usize) {
+        self.added_ticks.drop(cap, len);
+        self.changed_ticks.drop(cap, len);
+        self.data.drop(cap, len);
+        #[cfg(feature = "track_change_detection")]
+        self.changed_by.drop(cap, len);
+    }
+
+    /// Drops the last component in this column.
+    ///
+    /// # Safety
+    /// - `last_element_index` is indeed the index of the last element
+    /// - the data stored in `last_element_index` will never be used unless properly initialized again.
+    pub(crate) unsafe fn drop_last_component(&mut self, last_element_index: usize) {
+        std::ptr::drop_in_place(self.added_ticks.get_unchecked_raw(last_element_index));
+        std::ptr::drop_in_place(self.changed_ticks.get_unchecked_raw(last_element_index));
+        #[cfg(feature = "track_change_detection")]
+        std::ptr::drop_in_place(self.changed_by.get_unchecked_raw(last_element_index));
+        self.data.drop_last_element(last_element_index);
+    }
+
+    /// Get a slice to the data stored in this [`ThinColumn`].
+    ///
+    /// # Safety
+    /// - `T` must match the type of data that's stored in this [`ThinColumn`]
+    /// - `len` must match the actual length of this column (number of elements stored)
+    pub unsafe fn get_data_slice_for<T>(&self, len: usize) -> &[UnsafeCell<T>] {
```

Comment:
To be consistent with `get_added_ticks_slice`/`get_changed_ticks_slice` this should not end with `_for`